native/cocos/renderer/pipeline/PipelineSceneData.cpp

```diff
@@ -66,23 +66,16 @@ void PipelineSceneData::activate(gfx::Device *device) {
 }
 
 void PipelineSceneData::destroy() {
-    for (const auto &pair : _shadowFrameBufferMap) {
-        pair.second->destroy();
-        delete pair.second;
-    }
-
     _shadowFrameBufferMap.clear();
     _validPunctualLights.clear();
 
-    CC_SAFE_DESTROY_AND_DELETE(_occlusionQueryInputAssembler);
-    CC_SAFE_DESTROY_AND_DELETE(_occlusionQueryVertexBuffer);
-    CC_SAFE_DESTROY_AND_DELETE(_occlusionQueryIndicesBuffer);
+    _occlusionQueryInputAssembler = nullptr;
+    _occlusionQueryVertexBuffer = nullptr;
+    _occlusionQueryIndicesBuffer = nullptr;
 }
 
 void PipelineSceneData::initOcclusionQuery() {
-    if (!_occlusionQueryInputAssembler) {
-        _occlusionQueryInputAssembler = createOcclusionQueryIA();
-    }
+    _occlusionQueryInputAssembler = createOcclusionQueryIA();
```

Comment:
We might `CC_ASSERT(!_occlusionQueryInputAssembler)` here.

Reply:
It is better to add asset.